### PR TITLE
Implement option to use queue or cron (default) for webhook notification processing

### PR DESCRIPTION
**Pull request by factorin-j**
**Description**
Add configuration to use queue for async processing for webhook notifications or use default cron process.

**Tested scenarios**

Fixes  https://github.com/Adyen/adyen-magento2/issues/1847

### CODE REVIEW

**Comment by Morerice** [comment]
Hi @factorin-j,

Thanks for creating this PR! We will check it out and leave any feedback we have (if any) soon.

Thanks,
Jean
Adyen

File: Model/Queue/Notification/Consumer.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /** @var AdyenLogger $adyenLogger */
+    private $adyenLogger;
+
+    /**
+     * @param Webhook $webhookHelper
+     * @param AdyenLogger $adyenLogger
+     */
+    public function __construct(
+        Webhook $webhookHelper,
+        AdyenLogger $adyenLogger,
```
Comment (candemiralp) [inline Model/Queue/Notification/Consumer.php:26]:
```suggestion
        AdyenLogger $adyenLogger
```

Comment (factorin-j) [reply Model/Queue/Notification/Consumer.php]: Hi @candemiralp, 

Thanks! I have updated the PR and apply the suggested changes.

Comment (candemiralp) [reply Model/Queue/Notification/Consumer.php]: Hi @factorin-j,

Thank you for your valuable contribution. We will merge it after we complete the testing and final reviews.

Best Regards,
Can

**Review by candemiralp — APPROVED** [review]

**Review by Morerice — APPROVED** [review]

File: Model/Queue/Notification/Consumer.php (hunk 1 above)
Comment (factorin-j) [reply Model/Queue/Notification/Consumer.php]: Hi @candemiralp, 

Great! Thanks

**Comment by othuress** [comment]
Hi @factorin-j @candemiralp @Morerice 
This for sure is peace of xxx code and PR, we just upgraded magento and adyen to the latest version. NOTHING works, we have been using crons scheduled on db for like 10 years and suddenly adyen plugin stops working becuase of this and we do not find any way to fix this. 

[2023-04-04 12:53:16] AdyenLoggerTest.WARNING: Unknown connection name amqp

we have tried to disable amqp, change 'cron_consumers_runner' in env.php and Disable webapisync.

This PR need some serious rework and testing before release, becuase our entire operation is at full STOP!

**Comment by othuress** [comment]
I was a bit too fast, the webapiasync did the trick, but raises the question to why we get this error. I mean even though we use db scheduling their can be async work going on without RabbitMQ or any other broker actually being active on the instance. The PR should consider that in the build.

**Comment by factorin-j** [comment]
Hi @othuress, I'd like to confirm if you set the webhook notification processor to `Queue`? 
Screenshot 2023-04-07 at 2 54 09 AM

I can see the warning message reported when you choose queue (instead of cron) if you don't have RabbitMQ installed. 
`[2023-04-06T18:58:03.208068+00:00] AdyenLoggerTest.WARNING: Unknown connection name amqp`

If you do not have RabbitMQ installed, `Cron` should be selected as the processor to use the default behavior.

**Comment by othuress** [comment]
Hi @factorin-j , well we have never used Rabbit MQ and never installed it or enabled it. We have runing normal crons for over 10 years in Magento. Suddenly getting this kind of error seemed strange, and when we found out/were told that magento native is shipped with both but Rabbit Mq has never been enabled, we tried to force cron in the env.php that we did not have before the upgrade to adyen 8.x where this check for message que were introduced. adding this to env.php did not help so the solution was to disable async execution of cron jobs, then we stopped getting this error message. We have been using Cron setting in Adyen always and never changed that, not even sure that settings existed before the upgrade to Adyen 8.x

**Comment by factorin-j** [comment]
Hi @othuress, can you verify that `Cron` is selected as your processor? Once set, you should not be receiving this warning message even without disabling WebApiAsync module. This will use the default behavior on how Adyen usually processes notifications in the previous releases. Kindly let me know if setting it to cron fixes your issue. 

If you are receiving `Unknown connection name amqp` message, then it tells me that you have enabled queue instead of cron. By default, cron should be selected unless you set your store configuration to use queue.

This option is recently added in release version [8.15.0](https://github.com/Adyen/adyen-magento2/releases/tag/8.15.0)

**Comment by othuress** [comment]
@factorin-j Yes I can confirm the Cron setting as our processor. No we have not enabled que. No setting it to cron was not the game changer, it was already set to cron. There is a bugg in Adyen plugin that needs to be addressed.

the solution was turning the WebApiAsync module of, and that is a draw back for us in general Adyen forcing us to turn this of.

**Comment by factorin-j** [comment]
@othuress I think it would be better to create an issue and assign it me so I can investigate further. Kindly provide the following as well.

- Magento version
- Adyen version
- Full log line (including the stacktrace, if possible)

**Comment by othuress** [comment]
Magento version 2.4.3-p2
Adyen 8,16

I do not have a stack trace at this moment since we moved servers and newrelic was not installed before the upgrade.

Vänliga hälsningar / Mit freundlichen Grüßen / Best regards​​​​
Ola Thuresson
Direct. +46-(0)42-3017460<tel:+46-(0)42-3017460>
/
Cell. +46-(0)708-336040<tel:+46-(0)708-336040>
E‑mail. ***@***.******@***.***>

Thuresson Trading
Görslövsvägen 20
263 71
Jonstorp
Sweden
Switch. +46‑(0)42‑36 75 20
***@***.***<https://www.thuressontrading.com/>

**Comment by factorin-j** [comment]
@othuress unfortunately, I can't replicate the issue through a default installation with the same versions you have.
- PHP 7.4
- Magento 2.4.3-p2
- Adyen 8.16.0
- No RabbitMQ installed

I can only see the warning message if I change the Adyen's webhook notification processor to `queue` 
```
[2023-04-12 21:51:08] AdyenLoggerTest.WARNING: Unknown connection name amqp
#0 /var/www/magento/vendor/magento/framework-message-queue/PublisherPool.php(143): Magento\Framework\MessageQueue\ConnectionTypeResolver->getConnectionType('amqp')
#1 /var/www/magento/vendor/magento/framework-message-queue/PublisherPool.php(88): Magento\Framework\MessageQueue\PublisherPool->getPublisherForConnectionNameAndType('async', 'amqp')
#2 /var/www/magento/vendor/adyen/module-payment/Model/Queue/Notification/Publisher.php(29): Magento\Framework\MessageQueue\PublisherPool->publish('adyen.notificat...', Object(Adyen\Payment\Model\Notification))
#3 /var/www/magento/vendor/adyen/module-payment/Cron/WebhookProcessor.php(130): Adyen\Payment\Model\Queue\Notification\Publisher->execute(Object(Adyen\Payment\Model\Notification))
#4 /var/www/magento/vendor/adyen/module-payment/Cron/WebhookProcessor.php(84): Adyen\Payment\Cron\WebhookProcessor->doProcessWebhook()
#5 [internal function]: Adyen\Payment\Cron\WebhookProcessor->execute(Object(Magento\Cron\Model\Schedule))
```

You can verify it in the Adyen configuration in `Stores` > `Configuration` > `Sales` > `Payment Methods` > `Adyen` > `Advanced settings` > `Magento Order Processing` > `Webhook Notification Processor`
Or check it from `core_config_data` table; the value should be `cron` or an empty set.

I suggest to open a Github issue and include the stack trace of the warning message, as this will help us to pinpoint the root cause. It should be available in `var/log/adyen/warning.log` from our project's root directory.

**Pull request merged**